fix: promise support with webFrameMain.executeJavaScript

deepak1556 opened the pull request:
#### Description of Change

Closes https://github.com/electron/electron/issues/32756.
Closes https://github.com/electron/electron/issues/32164.

PR changes the backend for `WebFrameMain::ExecuteJavaScript` to use the version that allows resolving promises before returning the result. Not sure why the previous version which does not support promises https://source.chromium.org/chromium/chromium/src/+/main:content/browser/renderer_host/render_frame_host_impl.cc;l=2667-2670 was adopted in https://github.com/electron/electron/commit/704d69a8f96d904afc725f121432d23b7f44bd33 but this aligns the behavior with `WebFrame::ExecuteJavaScript`.

Was identified when writing tests for https://github.com/electron/electron/pull/35281

#### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: fix enable promise support with webFrameMain.executeJavaScript

samuelmaddock commented on shell/browser/api/electron_api_web_frame_main.cc:
```diff
@@ -141,17 +142,21 @@ v8::Local<v8::Promise> WebFrameMain::ExecuteJavaScript(
+                    blink::mojom::JavaScriptExecutionResultType::kSuccess)
+                  promise.Resolve(value);
+                else
+                  promise.Resolve(base::Value());
```
Can we reject the promise in the case of an error? Then we can match the behavior of WebContents.executeJavaScript.

codebytere replied:
we should also add a test for this case too

samuelmaddock reviewed (changes requested):
Good find! I looked into this a couple months ago and I think the promise resolution behavior is relatively new, added after webFrameMain was introduced.

deepak1556 commented:
Changed from `semver/patch` to `semver/minor` since there is change in functionality that api now rejects when result is a failure.

Also removed `target/19-x-y` since it requires https://chromium-review.googlesource.com/c/chromium/src/+/3651751 to be backported.

samuelmaddock reviewed (approved):
API LGTM

jkleinsc reviewed (approved):
API LGTM

MarshallOfSound commented:
> since there is change in functionality that api now rejects when result is a failure.

I'd argue that is reasonably expected behavior, a promise should never be assumed to _never_ reject 🤷

zcbenz reviewed (approved):
API LGTM

deepak1556 commented:
@MarshallOfSound the previous API always resolved to an empty `{}` even when the script execution fails, given now we resurface the actual error, I was thinking a `semver/minor` would be appropriate for this change rather than a patch, thoughts ?
